Assets/_Project/Scripts/Map/LocationFactory.cs

```diff
@@ -0,0 +1,81 @@
+using System.Collections;
+using System.Collections.Generic;
+using Mapbox.Unity.Location;
+using UnityEngine;
+
+namespace PolSl.UrbanHealthPath.Map
+{
+    public class LocationFactory
+    {
+        private LocationFactoryMode _mode;
+
+        private List<string> _latitudeLongitudeList;
+        
+        private ILocationProvider _locationProvider;
+        
+        private Location _currentLocation;
+        
+        public LocationFactory(LocationFactoryMode mode)
```

Comment:
A factory is a class that based on provided data returns an instance of an object.
This class does way more, so SRP violation.
I'll PM you with more details on how to refactor this class.